pkg/runtime/wfengine/wfengine.go

```diff
@@ -46,38 +43,61 @@ type WorkflowEngine struct {
 	actorRuntime   actors.Actors
 	startMutex     sync.Mutex
 	disconnectChan chan any
+	config         *WFConfig
 }
 
 var (
 	wfLogger            = logger.NewLogger("dapr.runtime.wfengine")
 	errExecutionAborted = errors.New("execution aborted")
 )
 
+// WFConfig is the configuration for the workflow engine
+type WFConfig struct {
+	AppID             string
+	WorkflowActorType string
+	ActivityActorType string
+}
+
 func IsWorkflowRequest(path string) bool {
 	return backend.IsDurableTaskGrpcRequest(path)
 }
 
-func NewWorkflowEngine() *WorkflowEngine {
+func NewWorkflowEngine(config *WFConfig) *WorkflowEngine {
+	actorTypePerAppPrefix := strings.Builder{}
+	actorTypePerAppPrefix.WriteString(actors.InternalActorTypePrefix)
+	actorTypePerAppPrefix.WriteString(utils.GetNamespaceOrDefault())
+	actorTypePerAppPrefix.WriteString(utils.DotDelimiter)
+	actorTypePerAppPrefix.WriteString(config.AppID)
+	actorTypePerAppPrefix.WriteString(utils.DotDelimiter)
+	actorTypePerAppPrefixStr := actorTypePerAppPrefix.String()
```

Comment:
This manual string-building seems like unnecessary complexity. I would expect something like this for extremely performance-sensitive code. However, this is *not* performance sensitive (we initialize the engine exactly once) so I find this kind of code to be distracting and unnecessarily hard to understand.

Reply:
Removed